From d16efb52126f790df60434261b9bc8cd8e6cf96c Mon Sep 17 00:00:00 2001
From: Ramesh Babu <ramesh.babu@intel.com>
Date: Mon, 6 Jun 2016 13:13:15 +0530
Subject: [PATCH 3996/4301] ALSA: hda: ext: add spib to stream context.

commit a52baaadad734288b8586e5c83ac1bfdbbbc8b8e from
git://git@github.com:01org/linux-intel-4.9.git

Platforms like skylake support SPIB (software position index in
Buffer) capability, through which application pointer can be
programmed in DMA. This helps DMA stop rendering stale data.

This patch saves spib values in stream context which can be
restored during resume from S3.

Change-Id: I2992242087ee0732b6fc571b5e65eb59aa1fa251
Signed-off-by: Ramesh Babu <ramesh.babu@intel.com>
Signed-off-by: Subhransu S. Prusty <subhransu.s.prusty@intel.com>
Signed-off-by: Mallikarjun, chippalkatti <mallikarjunx.chippalkatti@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/15038
Reviewed-by: audio_build <audio_build@intel.com>
Reviewed-by: Sm, Bhadur A <bhadur.a.sm@intel.com>
Tested-by: Sm, Bhadur A <bhadur.a.sm@intel.com>
---
 include/sound/hdaudio_ext.h     |    1 +
 sound/hda/ext/hdac_ext_stream.c |    2 ++
 2 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/include/sound/hdaudio_ext.h b/include/sound/hdaudio_ext.h
index fc530de..45dd76c 100644
--- a/include/sound/hdaudio_ext.h
+++ b/include/sound/hdaudio_ext.h
@@ -88,6 +88,7 @@ struct hdac_ext_stream {
 
 	u32 dpib;
 	u32 lpib;
+	u32 spib;
 	bool decoupled:1;
 	bool link_locked:1;
 	bool link_prepared;
diff --git a/sound/hda/ext/hdac_ext_stream.c b/sound/hda/ext/hdac_ext_stream.c
index 3b58210..f7116d9 100644
--- a/sound/hda/ext/hdac_ext_stream.c
+++ b/sound/hda/ext/hdac_ext_stream.c
@@ -460,6 +460,8 @@ int snd_hdac_ext_stream_set_spib(struct hdac_ext_bus *ebus,
 	}
 
 	writel(value, stream->spib_addr);
+	/* save the value in stream context */
+	stream->spib = value;
 
 	return 0;
 }
-- 
1.7.5.4

